### PR TITLE
chore(internal docs): Update RFC guidelines and template

### PR DESCRIPTION
**Pull request by binarylogic**

### CODE REVIEW

File: rfcs/README.md [hunk 1]
```diff
@@ -1,48 +1,122 @@
...
+3. Structure your RFC with headings that address one point at a time. This is necessary to facilitate productive
+   discussion via inline comments.
+4. Start with "why". Lead with one or two sentences describing the higher level intent. Leading with "how" omits the
+   opportunity to consider alternative approaches.
+
+### Keep the scope small
+
+Use the "Scope" section to address out of scope concerns, like future improvements. This signals to the team that
+you are aware of these concerns but have explicitly chosen to defer them. This not only helps to keep the discussion
+focused, but it
```
Comment (spencergilbert) [inline rfcs/README.md:68]: but it...?

**Review by jszwedko — APPROVED** [review]
Nice, yeah, I like this. I think it provides more clarity and guidelines. Thanks!

File: rfcs/README.md [hunk 2]
```diff
@@ -1,48 +1,123 @@
...
    * The change might be blocked by other work.
 
-### Creating a RFC
+### Creating an RFC
+
+1. Create a new branch
+1. Copy the [`rfcs/_YYYY-MM-DD-issue#-title.md`](rfcs/_YYYY-MM-DD-issue%23-title.md) template with the appropriate
+   name. Be sure to use the issue number you created above. (e.g., `rfcs/2020-02-10-445-internal-observability.md`)
+1. Fill in your RFC, pay attention the bullets and guidelines. Do not ommit any sections.
+1. Work with your team to land on a confident solution. Allocate time for code-level spikes if necessary.
```
Comment (JeanMertz) [inline rfcs/README.md:97]: > Work with your team

Minor nit: open-source contributors aren't part of any "team".

File: rfcs/README.md [hunk 3]
```diff
@@ -1,48 +1,123 @@
...
+1. Submit your RFC as a pull request and tag reviewers for approval.
+
+### Accepting an RFC
+
+1. At least 3 team members must approve your RFC in the form of pull request approvals.
+1. Once approved, self-merge your RFC.
```
Comment (JeanMertz) [inline rfcs/README.md:103]:
```suggestion
1. Once approved, self-merge your RFC, or ask someone to do it for you.
```

File: rfcs/README.md [hunk 4]
```diff
@@ -1,48 +1,123 @@
...
+
+### Implementing an RFC
+
+1. Create issues from the "Plan Of Attack" section. Place them in an epic if necessary.
+1. Coordinate with leadership to schedule your work.
```
Comment (JeanMertz) [inline rfcs/README.md:108]: This too doesn't fully apply to open-source contributors.

I'm fine leaving it as is, but these _are_ steps that non-DD employees can't necessarily take.

**Review by JeanMertz — APPROVED** [review]
I dig it. Left a few remarks, but nothing major.

**Comment by binarylogic** [comment]
I had a chat with @jszwedko and @StephenWakely about the issues I raised above. They sold me on the resources abstraction, but I want to document the details since I'd like to see these represented in the RFC:

* Like the services that sources and sinks integrate with, enrichment "resources" will likely need administrator approval and setup. Typically data is enriched from some sort of shared company "resource" that end users will likely not have access to.
* As an extension of the above requirement, admin will want to restrict access to resources to approved pipelines, especially if sensitive information is contained.
* Even though a CSV file is a "resource", it should not restrict how we interact with other resource types, like databases. When we add database support, we should have full reign to solve that however we see fit. For example, we might deem using the database connection sufficient versus loading the data into an in-memory cache.

A few other RFC details that we decided on:

* We should note encryption but not solve it in this RFC. In other words, lets makes space for decryption and note where the parameters will be specified, but refrain from solving that in this RFC. We can note encryption as a follow up item in the "Plan of attack".
* We should try out best to break this down into small incremental steps. For example, adding support for resources first, the VRL function second, encryption last. I think we could break it down even more than this, but at the minimum it should be broken down into those steps.

**Pull request merged**